services/wallet/router/pathprocessor/processor_erc721.go

```diff
@@ -37,23 +37,41 @@ type ERC721TxArgs struct {
 }
 
 type ERC721Processor struct {
-	rpcClient  *rpc.Client
-	transactor transactions.TransactorIface
+	rpcClient         *rpc.Client
+	transactor        transactions.TransactorIface
+	excludedContracts []common.Address
 }
 
-func NewERC721Processor(rpcClient *rpc.Client, transactor transactions.TransactorIface) *ERC721Processor {
-	return &ERC721Processor{rpcClient: rpcClient, transactor: transactor}
+func NewERC721Processor(rpcClient *rpc.Client, transactor transactions.TransactorIface, excludedContracts []common.Address) *ERC721Processor {
```

Comment:
contracts should be identified by chain+Address
We actually have type ContractID :D
<img width="552" height="154" alt="image" src="https://github.com/user-attachments/assets/c799fc46-d43e-4443-a049-549861bfcb7a" />

Reply:
@friofry if you decide to have a separate processor (instead of integrating it into erc721), then `excludedContracts` is something I would keep internal for the processor, not passing from outside.